src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.SourceBuild.Tasks.LeakDetection/CustomAttributeTypeProvider.cs

```diff
@@ -0,0 +1,85 @@
+using System;
+using System.Reflection;
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
+
+namespace Microsoft.DotNet.SourceBuild.Tasks.LeakDetection
+{
+    internal class CustomAttributeTypeProvider : ICustomAttributeTypeProvider<Type>
+    {
+        private readonly MetadataReader _reader;
+        public CustomAttributeTypeProvider(MetadataReader reader)
+        {
+            _reader = reader;
+        }
+
+        public Type GetPrimitiveType(PrimitiveTypeCode typeCode)
+        {
+            switch (typeCode)
+            {
+            case PrimitiveTypeCode.Boolean:
+                return typeof(bool);
+            case PrimitiveTypeCode.Byte:
+                return typeof(byte);
+            case PrimitiveTypeCode.Int32:
+                return typeof(int);
+            case PrimitiveTypeCode.String:
+                return typeof(string);
+            default:
+                throw new NotImplementedException();
```

Comment:
These lines should be indented. Also applicable to another `switch` below.

Reply:
Updated in https://github.com/dotnet/installer/pull/17605/commits/2e6190436a7326641913a147d0afe7d505eefef5